sled-agent/src/instance_manager.rs

```diff
@@ -686,12 +687,11 @@ impl InstanceManagerRunner {
 
         instance
             .issue_snapshot_request(tx, disk_id, snapshot_id)
-            .await
             .map_err(Error::from)
     }
 
     /// Create a zone bundle from a named instance zone, if it exists.
-    async fn create_zone_bundle(
+    fn create_zone_bundle(
```

Comment:
While working on this change, I noticed that the error handling in the rest of this function is a little bit off. We presently return errors from `create_zone_bundle` if the provided zone name is invalid, or if there's no instance for that UUID. These errors are bubbled up into the `InstanceManager` run loop, which logs an error...but the response `tx` oneshot just gets dropped, rather than receiving these errors.

This means that the client gets a generic "response dropped" error rather than the more useful error indicating that the zone name is bad or doesn't exist. We should change this function so the errors are sent back over the channel instead, but I felt like that was better done in a separate PR from the rest of this change, for Git hustory reasons